data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py

```diff
@@ -0,0 +1,268 @@
+        # equal probability into continuous interval. I opted for cut as it offers greater flexibility in
+        # creating percentile bins.
+        #
+        # The result produces a cateogrical variable that affords easier stratification
+        # when performing more advanced statistical analyses of each component of the distribution. This is pertienent
+        # if we seek to combine this data with population characteristics and demographics - for example, ACS and CDC SVI Index -
+        # in order to chracterize each section of the "higher scoring" tracts for cancer-related and non-cancer related toxicity
+        self.df[field_names.EPA_RSEI_SCORE_PERCENTILE_LABEL_FIELD] = pd.cut(
+            self.df[field_names.EPA_RSEI_SCORE_OUTPUT_FIELD],
+            right=True,  # include right-most interval
+            bins=perc_func_overall_risk_score,
+            labels=labels_for_cut_function,
+            ordered=True,
+        )
+
+        self.df[field_names.EPA_RSEI_CSCORE_PERCENTILE_LABEL_FIELD] = pd.cut(
+            self.df[field_names.EPA_RSEI_CSCORE_OUTPUT_FIELD],
+            right=True,
+            bins=perc_func_cancer_score,
+            labels=labels_for_cut_function,
+            ordered=True,
+        )
+
+        self.df[field_names.EPA_RSEI_NCSCORE_PERCENTILE_LABEL_FIELD] = pd.cut(
+            self.df[field_names.EPA_RSEI_NCSCORE_OUTPUT_FIELD],
+            right=True,
+            bins=perc_func_non_cancer_score,
+            labels=labels_for_cut_function,
+            ordered=True,
+        )
+
+        # Produce percentile rank for overall risk score
+        # This was created for the sole purpose to be used in the current
+        # iteration of Score L
+        self.df[field_names.EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD] = self.df[
+            field_names.EPA_RSEI_SCORE_OUTPUT_FIELD
+        ].rank(
+            ascending=True,
+            pct=True,
+        )
+
+        # This threshold was arbitrarily chosen.
+        # It would make sense to enrich this with facilities, industries, or chemical
+        # that would enable some additional form of sub-stratification when examining
+        # different percentile ranges that are derived above.
+        self.df[field_names.EPA_RSEI_SCORE_THRESHOLD_FIELD] = (
+            self.df[field_names.EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD] > 0.75
+        )
+
+        expected_census_tract_field_length = 11
+        self.df[self.GEOID_TRACT_FIELD_NAME] = (
+            self.df[self.GEOID_TRACT_FIELD_NAME]
+            .astype(str)
+            .apply(lambda x: x.zfill(expected_census_tract_field_length))
+        )
+
+        if (
+            not self.df[self.GEOID_TRACT_FIELD_NAME]
+            .apply(lambda x: len(str(x)))
+            .eq(expected_census_tract_field_length)
+            .all()
+        ):
+            raise ValueError(
+                f"GEOID Tract must be length of {expected_census_tract_field_length}"
```

Comment:
We seem to reuse this same check in many data source ETL classes. Does it make sense to delegate this to the superclass?

Reply:
yea definitely!

Reply:
This is a good example of where `pandera` can be used to enforce a contract across all ETL classes simply by invoking a superclass's utility. I will be working with @sverchdotgov on this today.

Reply:
This comment did not age well - as I said, as OS contributor I will continue to this project and will address this even after tomorrow. I care about this team's work.